pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -323,6 +418,80 @@ func configMapDeleted(obj interface{}) {
+		log.Errorf("fssCRAdded: failed to cast object to %s. err: %v", featurestates.CRDName, err)
+		return
+	}
+	if svFSSObject.Name != featurestates.SVFeatureStateCRName {
+		log.Warnf("fssCRAdded: Ignoring %s CR object with name %q", featurestates.CRDName, svFSSObject.Name)
+		return
+	}
+	for _, fss := range svFSSObject.Spec.FeatureStates {
+		k8sOrchestratorInstance.supervisorFSS.featureStates[fss.Name] = strconv.FormatBool(fss.Enabled)
+	}
+	log.Infof("fssCRAdded: New supervisor feature states values stored successfully from %s CR object: %v",
+		featurestates.SVFeatureStateCRName, k8sOrchestratorInstance.supervisorFSS.featureStates)
+}
+
+// fssCRUpdated updates supervisor feature state switch values from the cnscsisvfeaturestate CR
+func fssCRUpdated(oldObj, newObj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	var svFSSObject featurestatesv1alpha1.CnsCsiSvFeatureStates
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(newObj.(*unstructured.Unstructured).Object, &svFSSObject)
+	if err != nil {
+		log.Errorf("fssCRUpdated: failed to cast object to %s. err: %v", featurestates.CRDName, err)
+		return
+	}
+	if svFSSObject.Name != featurestates.SVFeatureStateCRName {
+		log.Warnf("fssCRUpdated: Ignoring %s CR object with name %q", featurestates.CRDName, svFSSObject.Name)
+		return
+	}
+	for _, fss := range svFSSObject.Spec.FeatureStates {
+		k8sOrchestratorInstance.supervisorFSS.featureStates[fss.Name] = strconv.FormatBool(fss.Enabled)
+	}
+	log.Infof("fssCRUpdated: New supervisor feature states values stored successfully from %s CR object: %v",
+		featurestates.SVFeatureStateCRName, k8sOrchestratorInstance.supervisorFSS.featureStates)
+}
+
+// fssCRDeleted marks feature state switch values for all features in supervisor feature state
+// switch to false and logs an error
+func fssCRDeleted(obj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	var svFSSObject featurestatesv1alpha1.CnsCsiSvFeatureStates
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &svFSSObject)
+	if err != nil {
+		log.Errorf("fssCRDeleted: failed to cast object to %s. err: %v", featurestates.CRDName, err)
+		return
+	}
+	if svFSSObject.Name != featurestates.SVFeatureStateCRName {
+		log.Warnf("fssCRDeleted: Ignoring %s CR object with name %q", featurestates.CRDName, svFSSObject.Name)
+		return
+	}
+	// Marking all FSS values as false
+	for _, fss := range svFSSObject.Spec.FeatureStates {
+		k8sOrchestratorInstance.supervisorFSS.featureStates[fss.Name] = strconv.FormatBool(false)
+	}
+	// Logging an error here because cnscsisvfeaturestate CR should not be deleted
+	log.Errorf("fssCRDeleted: %s CR object with name %q deleted", featurestates.CRDName, svFSSObject.Name)
```

Comment:
2 approaches:
1. Do not mark FSS as false. Crash container
2. Keep stale FSS. Do not register the delete event.

Reply:
We are going ahead with approach 1 after a discussion with the team.